### PR TITLE
Cosmos: *Properties related code and doc changes

### PR DESCRIPTION
**Pull request by milismsft**
Implement changes captured in https://github.com/Azure/azure-sdk-for-java/issues/11483

- Update documentation for getting the request charges of a Cosmos operation.
- Update getResourceId() to be package private (it was public).
- Update *Properties for Cosmos scripts to package private; add constructor that takes id and body. Update respective tests.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/UserQueryTest.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ public void queryAllUsers() throws Exception {
 
         FeedResponseListValidator<CosmosUserProperties> validator = new FeedResponseListValidator.Builder<CosmosUserProperties>()
                 .totalSize(expectedUsers.size())
-                .exactlyContainsInAnyOrder(expectedUsers.stream().map(d -> d.getResourceId()).collect(Collectors.toList()))
+                .exactlyContainsInAnyOrder(expectedUsers.stream().map(CosmosUserProperties::getId).collect(Collectors.toList()))
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/UserQueryTest.java:89]: We are replacing `getResourceId()` -> with `getId()`
Can you please make sure these tests pass ?
Here and everywhere else in your changes ?

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/UserQueryTest.java:89]: Please don't replace this in the tests. ResourceId is a unique value, however id may not necessary be unique. 

To access package private getResource you can use BridgeInternal if needed. please roll back these changes.

Comment (milismsft) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/UserQueryTest.java:89]: So basically what this comes to is that UUID.randomUUID().toString() does not generate unique IDs??? If yes than we have a bigger problem through the whole test bed.

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/UserQueryTest.java:89]: Milis, are you taking into account that ID may not necessarily be unique across multiple partitions?

Comment (milismsft) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/UserQueryTest.java:89]: The IDs for all those tests are generated using the API I just mentioned; they pretty much should be unique as they come out from it. The second thing is that there are no partition setting that are applicable to databases, containers, container scripts for instance. All these tests use the IDs to count the results (see Reactor collect() call following them).
So unless we have a special test that create a batch of documents and they all inserted in two or more sets with different partition key values, we should be fine.
I'm running the full set of tests; that should give us the validation we need, right?

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/UserQueryTest.java:89]: @milismsft  as the goal of the PR is to change the public api, I suggest we don't change how tests works. If you have thought on improving the tests please consider anything other than public api change outside of the PR for public api change.

as a pattern we rely on resourceId to ensure we cover having same id across multiple partition scenario.

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/UserQueryTest.java:89]: Please check the internal logic of  `FeedResponseListValidator#exactlyContainsInAnyOrder()` 
if you plan to use id for non item resource please consider using a different method.

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/UserQueryTest.java:89]: FeedResponseListValidator.exactlyContainsInAnyOrder uses resourceid to internally validation. 

now on one side you are passing id and on the other side it compares against resoruceId, I suspect the tests will fail.

**Pull request merged**